sparsesim/src/lib.rs

```diff
@@ -190,7 +190,7 @@ impl QuantumSim {
     /// Prints the current state vector to standard output with integer labels for the states, skipping any
     /// states with zero amplitude.
     #[allow(clippy::missing_panics_doc)] // reason="Panics can only occur if the keys are not present in the map, which should not happen."
-    pub fn dump(&mut self) {
+    pub fn dump(&mut self) -> String {
```

Comment:
This gets used in a few tests just to aid in visually inspecting behavior, so I think it might be worth having `println!` around those now. Maybe adding a `[must_use]` attr would help identify those, but that's up to you if you want to go that far.